packages/material-ui/src/ListItem/ListItem.js

```diff
@@ -107,11 +109,23 @@ const ListItem = React.forwardRef(function ListItem(props, ref) {
     dense: dense || context.dense || false,
     alignItems,
   };
+  const listItemRef = React.useRef();
+  React.useLayoutEffect(() => {
+    if (autoFocus) {
+      listItemRef.current.focus();
```

Comment:
I think we had some instances where `focus` wasn't a function in IE11. Should we check for those? Maybe 
```suggestion
      const { focus = () => warning(false, "where's my focus at? :(") } = listItemNode;
      focus();
```

which had a guard before:
```js
const { current: listItemNode } = listItemRef;
if (listItemNode == null) return;
```

Reply:
@eps1lon From what I remember, it's an issue on IE 11 with svg elements. We should blame to get more context:
```js

        // Not all elements in IE 11 have a focus method.
        // Because IE 11 market share is low, we accept the restore focus being broken
        // and we silent the issue.
        if (lastFocus.current.focus) {
          lastFocus.current.focus();
        }
```

Reply:
It's probably safe to ignore here: https://github.com/mui-org/material-ui/pull/11115.